server/protocols/webadmin/webadmin-rabbitmq/src/main/java/org/apache/james/webadmin/service/RebuildMailQueueTask.java

```diff
@@ -0,0 +1,83 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.service;
+
+import java.time.Instant;
+import java.util.Optional;
+
+import org.apache.james.queue.rabbitmq.RabbitMQMailQueue;
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskExecutionDetails;
+import org.apache.james.task.TaskType;
+
+
+public class RebuildMailQueueTask implements Task {
+
+    public static class AdditionalInformation implements TaskExecutionDetails.AdditionalInformation {
+
+        private final Instant timestamp;
+        private final Instant olderThan;
```

Comment:
The queue name needs to be part of the infos.

Also we should have a count of re-published messages (to not be blind).

Reply:
The queue name is useful since we can only perform it on one queue.

Reply:
It doesn't make sense to do it on another queue. no ?

Reply:
Sorry but if i launch this task 2 time, one time on spool, one time on outgoing, i need to know which one is what.

Reply:
You cannot run it on outgoing, only spool can be rebuild.

Reply:
Why?